scipy/misc/common.py

```diff
@@ -201,3 +202,100 @@ def face(gray=False):
+
+    The returned signal is a 5 minute long electrocardiogram (ECG), a medical
+    recording of the hearts electrical activity, sampled at 360 Hz.
+
+    Returns
+    -------
+    ecg : ndarray
+        The electrocardiogram in millivolt (mV) sampled at 360 Hz.
+
+    Notes
+    -----
+    The provided signal is an excerpt (19:35 to 24:35) from the `record 208`_
+    (lead MLII) provided by the MIT-BIH Arrhythmia Database [1]_ on
+    PhysioNet [2]_. The excerpt includes noise induced artifacts, typical
+    heartbeats as well as pathological changes.
+
+    .. _record 208: https://physionet.org/physiobank/database/html/mitdbdir/records.htm#208
+
+    .. versionadded:: 1.1.0
+
+    References
+    ----------
+    .. [1] Moody GB, Mark RG. The impact of the MIT-BIH Arrhythmia Database.
+           IEEE Eng in Med and Biol 20(3):45-50 (May-June 2001).
+           (PMID: 11446209); doi:10.13026/C2F305
+    .. [2] Goldberger AL, Amaral LAN, Glass L, Hausdorff JM, Ivanov PCh,
+           Mark RG, Mietus JE, Moody GB, Peng C-K, Stanley HE. PhysioBank,
+           PhysioToolkit, and PhysioNet: Components of a New Research Resource
+           for Complex Physiologic Signals. Circulation 101(23):e215-e220;
+           Circulation Electronic Pages: http://circ.ahajournals.org/content/101/23/e215.full];
+           2000 (June 13).; doi: 10.1161/01.CIR.101.23.e215
+
+    Examples
+    --------
+    >>> from scipy.misc import electrocardiogram
+    >>> ecg = electrocardiogram()
+    >>> ecg.shape, ecg.mean(), ecg.std()
+    ((108000,), -0.16510875, 0.5992473991177294)
+
+    As stated the signal features several areas with a different morphology.
+
+    >>> import matplotlib.pyplot as plt
+    >>> fs = 360
+    >>> time = np.arange(ecg.size) / fs
+    >>> plt.plot(time, ecg)
+    >>> plt.xlabel("time in s")
+    >>> plt.ylabel("ECG in mV")
+    >>> plt.xlim(9, 10.2)
+    >>> plt.ylim(-1, 1.5)
+    >>> plt.show()
+
+    The first few seconds show a the electrical activity of a heart in normal
+    sinus rhythm as seen above.
+
+    >>> plt.plot(time, ecg)
+    >>> plt.xlabel("time in s")
+    >>> plt.ylabel("ECG in mV")
+    >>> plt.xlim(46.5, 50)
+    >>> plt.ylim(-2, 1.5)
+    >>> plt.show()
+
+    After second 16, the first premature ventricular contractions, also called
+    extrasystoles, appear. These have a different morphology compared to typical
+    heartbeats which can be easily observed in the plot above.
+
+    >>> plt.plot(time, ecg)
+    >>> plt.xlabel("time in s")
+    >>> plt.ylabel("ECG in mV")
+    >>> plt.xlim(207, 215)
+    >>> plt.ylim(-2, 3.5)
+    >>> plt.show()
+
+    At several points large artifacts disturb the recording.
+
+    >>> from scipy.signal import welch
+    >>> f, Pxx = welch(ecg, fs=fs, nperseg=2048, scaling="spectrum")
+    >>> plt.semilogy(f, Pxx)
+    >>> plt.xlabel("Frequency in Hz")
+    >>> plt.ylabel("Power spectrum of the ECG in mV**2")
+    >>> plt.show()
```

Comment:
> If I'm not mistaken, they are missing plt.figure() lines at the beginning of each block.

No sadly that doesn't seem to be the cause. I'm not entirely sure what's going wrong but here is what I figured out so far:
Generating the plots fails if the docstring (or example section?) ends with the following pattern:
```
[...]
>>> plt.show()

Any text.
"""
```
It seems like the parsing of the docstring always fails for that specific pattern. Sphinx even prints a traceback of this, e.g.:
```
/home/lg/Projects/scipy/scipy/misc/common.py:docstring of scipy.misc.electrocardiogram:43: WARNING: Exception occurred in plotting scipy-misc-electrocardiogram-1
 from /home/lg/Projects/scipy/doc/source/generated/scipy.misc.electrocardiogram.rst:
Traceback (most recent call last):
  File "/home/lg/miniconda/envs/scipydev/lib/python3.6/site-packages/matplotlib/sphinxext/plot_directive.py", line 524, in run_code
    six.exec_(code, ns)
  File "<string>", line 2
    Examining the power spectrum reveals that most of the biosignal is made up
                ^
SyntaxError: invalid syntax
```
If I don't add a long sentence add the end but instead only a letter or word the syntax error turns into a `NameError: name 'Test' is not defined`. It seems to me like sphinx (or something else) tries to parse the line after a `plot.show()` as a python statement. If I add a valid statement 
 like `print(1)` the plots show up and the statement is included in the docstring as well. If I add just `p` the error reappears.

To reproduce, try adding this pattern to the docstring of `scipy.misc.face` which will show the same behavior. 

It might be important to mention that I'm building with Python 3.6 and replaced the line 32 in `doc/sphinxext/numpydoc/numpydoc.py` with `from docutils.parsers.rst import Directive` because `from sphinx.util.compat import Directive` raised an ImportError. I'm not sure if this is relevant. I give up for today and try to build / reproduce this tomorrow with Python 2.7.

Reply:
Same behavior with Python 2.7 and CircleCI seems to build with Python 3.6 anyway. I'm still a little bit confused why I needed to replace the import in `numpydoc` (as mentioned above) but that doesn't seem to be the cause.

I would try to file a bug report but I'm not sure where. Is it something in the setup of Scipy? A bug in numpydoc or sphinx? Might be worth it to come up with a dummy project to reproduce this bug.